spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/SimpleConnection.java

```diff
@@ -55,6 +56,7 @@ public Channel createChannel(boolean transactional) {
 		try {
 			Channel channel = this.delegate.createChannel();
 			if (transactional) {
+				Assert.state(channel != null, "Can't start the transaction - no channel is available.");
```

Comment:
??? 
Why did you decide to come back to this?
Gary asked you to check independently if the `transactional` state.

And I asked you to take a look to failing tests.
However it would be better to add a test-case to validate your fix.

Does it all make sense to you?